metron-platform/metron-enrichment/Performance.md

```diff
@@ -0,0 +1,527 @@
+	       }
+	     }
+	    },
+	    "fieldToTypeMap": {
+	    }
+	  },
+	  "threatIntel": {
+	  }
+	}
+	[Stellar]>>> CONFIG_PUT("ENRICHMENT", conf, "asa")
+	```
+
+### Create Input Telemetry
+
+1.  Create a template file that defines what your input telemetry will look-like.
+
+	```bash
+	> cat asa.template
+	{"ciscotag": "ASA-1-123123", "source.type": "asa", "ip_dst_addr": "$DST_ADDR", "original_string": "<134>Feb 22 17:04:43 AHOSTNAME %ASA-1-123123: Built inbound ICMP connection for faddr 192.168.11.8/50244 gaddr 192.168.1.236/0 laddr 192.168.1.1/161", "ip_src_addr": "192.168.1.35", "syslog_facility": "local1", "action": "built", "syslog_host": "AHOSTNAME", "timestamp": "$METRON_TS", "protocol": "icmp", "guid": "$METRON_GUID", "syslog_severity": "info"}
+	```
+
+2.  Use the template file along with the enrichment data to create input telemetry with varying IP addresses.
+
+	```bash
+	for i in $(head top-1m-with-ip.csv | awk -F, '{print $2}');do
+		cat asa.template | sed "s/\$DST_ADDR/$i/";
+	done > asa.input.template
+	```
+
+3. Use the `load_test.sh` script to push messages onto the input topic `enrichments` and monitor the output topic `indexing`.
+
+	If the topology is keeping up, obviously the events per second produced on the input topic should roughly match the output topic.
+
+	```
+	$METRON_HOME/bin/load_test.sh \
+		-e 200000 \
+		-ot enrichments \
+		-mt indexing \
+		-p 10 \
+		-t asa.input.template \
+		-z $ZOOKEEPER
+	```
+
+	[TODO] Link to the docs that get created for the `load_test.sh` script.
+
+### Cluster Setup
+
+#### Isolation
+
+The Enrichment topology depends on an environment with at least two and often three components that work together; Storm, Kafka, and HBase.  When any of two of these are run on the same node, it can be difficult to identify which of them is becoming a bottleneck.  This can cause poor and highly volatile performance as each steals resources from the other.  
+
+It is highly recommended that each of these systems be fully isolated from the others.  Storm should be run on nodes that are completely isolated from Kafka and HBase.
+
+### Monitoring
+
+1. The `load_test.sh` script will report the throughput for the input and output topics.  
+
+	* The input throughput should roughly match the output throughput if the topology is able to handle a given load.
+
+	* Not only are the raw throughput numbers important, but also the consistency of what is reported over time.  If the reported throughput is sporadic, then further tuning may be required.
+
+1. The Storm UI is obviously an important source of information.  The bolt capacity, complete latency, and any reported errors are all important to monitor
+
+1. The load reported by the OS is also an important metric to monitor.  
+
+	* The load metric should be monitored to ensure that each node is being pushed sufficiently, but not too much.
+
+	 * The load should be evenly distributed across each node.  If the load is uneven, this may indicate a problem.
```

Comment:
The additional space is putting this bullet into a code block when rendered in the site-book.